zebra-network/src/peer/connection.rs

```diff
@@ -585,7 +586,10 @@ where
                 .peer_tx
                 .send(Message::Mempool)
                 .await
-                .map_err(|e| e.into())
+                // TODO: work out how to re-use `tx`
+                // `tx` is only used once, in either the error or the result,
+                // but rustc doesn't know that
```

Comment:
@yaahc I don't know how to use tx in both `map` and `map_err`.

Should we leave the result as it is, and create the `AwaitingResponse` in the outer match?

Reply:
I don't think that will be necessary, if you just write out the `map` and `map_err` as a `match expr { Ok(e) => Ok(<expression>), Err(e) => Err(<expression>), }` it should be able to prove that the cases are mutually exclusive and will let you reuse the same variable in both.

Reply:
Fixed in 12fec92cc43e198127d567498809c215f96a57c4